src/NLog.Extensions.Logging/NLogLogger.cs

```diff
@@ -15,24 +16,19 @@ public NLogLogger(Logger logger)
         }
 
         //todo  callsite showing the framework logging classes/methods
-
-        public void Log(Microsoft.Extensions.Logging.LogLevel logLevel, int eventId, object state, Exception exception, Func<object, Exception, string> formatter)
+        public void Log<TState>(Microsoft.Extensions.Logging.LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
         {
             var nLogLogLevel = ConvertLogLevel(logLevel);
             if (IsEnabled(nLogLogLevel))
             {
-                string message;
-                if (formatter != null)
-                {
-                    message = formatter(state, exception);
-                }
-                else
+                if (formatter == null)
                 {
-                    message = Microsoft.Extensions.Logging.LogFormatter.Formatter(state, exception);
+                    throw new ArgumentNullException(nameof(formatter));
```

Comment:
I'm not sure, I followed the pattern now used in Console and Debug loggers...
https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging.Debug/DebugLogger.cs#L54-L64
https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging.Console/ConsoleLogger.cs#L86-L96

There is a huge commit transforming/removing LogFormatter...
https://github.com/aspnet/Logging/commit/4528bdbc2816a8b5f933d9c6cf2da806ce4a72ea